src/main/java/io/split/android/client/events/SplitEventsManager.java

```diff
@@ -0,0 +1,180 @@
+package io.split.android.client.events;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+
+import io.split.android.client.SplitClientConfig;
+import io.split.android.client.events.executors.SplitEventExecutorAbstract;
+import io.split.android.client.events.executors.SplitEventExecutorFactory;
+import io.split.android.client.events.executors.SplitEventExecutorResources;
+import io.split.android.client.utils.Logger;
+
+/**
+ * Created by sarrubia on 4/3/18.
+ */
+
+public class SplitEventsManager implements Runnable {
+
+    private final int QUEUE_CAPACITY = 10;
+
+    private SplitClientConfig _config;
+
+    private final ScheduledExecutorService _scheduler;
+
+    private ArrayBlockingQueue<SplitInternalEvent> _queue;
+
+    private Map<SplitEvent, List<SplitEventTask>> _suscriptions;
+
+
+
+    private SplitEventExecutorResources _resources;
+
+    private boolean _eventMySegmentsAreReady = false;
+    private boolean _eventSplitsAreReady = false;
+
+    private Map<SplitEvent, Integer> _executionTimes;
+
+    public SplitEventsManager(SplitClientConfig config){
+
+        _config = config;
+
+        _queue = new ArrayBlockingQueue<>(QUEUE_CAPACITY);
+        _suscriptions = new HashMap<>();
+
+        _executionTimes = new HashMap<>();
+        _resources = new SplitEventExecutorResources();
+
+        registerMaxAllowebExecutionTimesPerEvent();
+
+        Runnable SDKReadyTimeout = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    if (_config.blockUntilReady() > 0) {
+                        Thread.sleep(_config.blockUntilReady());
+                        notifyInternalEvent(SplitInternalEvent.SDK_READY_TIMEOUT_REACHED);
+                    }
+                } catch (InterruptedException e) {
+                    Logger.d(e.getMessage());
```

Comment:
Add some line to the logger explaining

Reply:
fire anyway

Reply:
also catch Throwable just in case, different catch, log different message, fire anyway

Reply:
this one too

Reply:
still